internal/pkg/vm/state/tree.go

```diff
@@ -41,45 +23,34 @@ type Tree interface {
 	SetActor(ctx context.Context, a address.Address, act *actor.Actor) error
 
 	ForEachActor(ctx context.Context, walkFn ActorWalkFn) error
+	GetAllActors(ctx context.Context) <-chan GetAllActorsResult
 }
 
-var _ Tree = &tree{}
-
-// IpldStore defines an interface for interacting with a hamt.CborIpldStore.
-// TODO #3078 use go-ipld-cbor export
-type IpldStore interface {
-	Put(ctx context.Context, v interface{}) (cid.Cid, error)
-	Get(ctx context.Context, c cid.Cid, out interface{}) error
-}
+// ActorWalkFn is a visitor function for actors
+type ActorWalkFn func(address.Address, *actor.Actor) error
 
-// TreeLoader defines an interfaces for loading a state tree from an IpldStore.
-type TreeLoader interface {
-	LoadStateTree(ctx context.Context, store IpldStore, c cid.Cid) (Tree, error)
+// GetAllActorsResult is the struct returned via a channel by the GetAllActors
+// method. This struct contains only an address string and the actor itself.
+type GetAllActorsResult struct {
+	Address string
+	Actor   *actor.Actor
+	Error   error
 }
 
-// TreeStateLoader implements the state.StateLoader interface.
-type TreeStateLoader struct{}
-
-// LoadStateTree is a wrapper around state.LoadStateTree.
-func (stl *TreeStateLoader) LoadStateTree(ctx context.Context, store IpldStore, c cid.Cid) (Tree, error) {
-	return LoadStateTree(ctx, store, c)
+// tree is a state tree that maps addresses to actors.
+type tree struct {
+	// root is the root of the state merklehamt
+	root  *hamt.Node
+	store *hamt.CborIpldStore
 }
 
-// LoadStateTree loads the state tree referenced by the given cid.
-func LoadStateTree(ctx context.Context, store IpldStore, c cid.Cid) (Tree, error) {
-	// TODO ideally this assertion can go away when #3078 lands in go-ipld-cbor
-	root, err := hamt.LoadNode(ctx, store.(*hamt.CborIpldStore), c, hamt.UseTreeBitWidth(TreeBitWidth))
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to load node for %s", c)
-	}
-	stateTree := newEmptyStateTree(store.(*hamt.CborIpldStore))
-	stateTree.root = root
-
-	return stateTree, nil
-}
+const (
+	// TreeBitWidth is the bit width of the HAMT used to store a state tree
+	TreeBitWidth = 5
```

Comment:
This constant should be moved back to the top of the file. This is inconsistent with how constants are defined throughout the codebase. 

Most of the code reordering in this file is unnecessary. There is no code style for how elements get ordered in this project. If we decide to implement a style for order of elements, we should make it explicit and implement and fix the codebase in a single PR. Piecemeal changes like just increase entropy and destroy git history.